include/CL/sycl/id.hpp

```diff
@@ -57,9 +57,9 @@ class id : public detail::small_array_123<
       { rhs.get_id() }
   {}
 
-  /// Keep other constructors
-  id() = default;
-
+  /// Default constructor must 0 all elements
+  id() : detail::small_array_123<std::size_t, id<Dimensions>, Dimensions> { 0 }
```

Comment:
I tried by undoing it but keeping your test and it seems to work.
But by looking at the C++ spec it is not clear to me if it is required or not and if the result is UB but just working... :-/
Do you have some failing cases in the CTS?
Anyway we can go with your change. since it is clearer.

Reply:
normally allocs aren't zeroed so you have to explicitly do it, the spec for ids does it, the spec doesn't do it for ranges. Yes there is a CTS test for it we were failing.